src/capellambse/diagram_cache.py

```diff
@@ -146,6 +146,8 @@ def _main(
             background=background,
             refresh=refresh,
         )
+        if any(not i["success"] for i in diagrams):
+            raise SystemExit(2)
```

Comment:
Looks like exit code 2 is already reserved for a wrong usage of the CLI itself.

When I command

```shell
python -m capellambse.diagram_cache
```
I get the output

```
Usage: python -m capellambse.diagram_cache [OPTIONS]
Try 'python -m capellambse.diagram_cache --help' for help.

Error: Missing option '-m' / '--model'.
```
with an exit code of 2. But that would be misleading.

I found that there are exit codes with special meanings:

https://tldp.org/LDP/abs/html/exitcodes.html#EXITCODESREF

This is why we should use something else and I propose to go for exit code 3.

Reply:
Other than that from above the `SystemExit` works. I checked it with a good real life example that led to this PR.
So when an exit code between three and 125 is chosen, I am happy with the changes in this PR.